testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -1903,6 +1904,66 @@ public void testDeleteAfterMajorCompaction()
         });
     }
 
+    @Test
+    public void testUnbucketedPartitionedTransactionalTableWithTaskWriterCountGreaterThan1()
+    {
+        withTemporaryTable("test_delete_with_unbucketed_transactional_table", true, false, NONE, tableName -> {
+            onTrino().executeQuery(format("CREATE TABLE %s  WITH (format='ORC', transactional=true, partitioned_by = ARRAY['orderpriority']) AS SELECT orderkey, orderstatus, totalprice, orderdate, clerk, shippriority, \"comment\", custkey, orderpriority FROM tpch.sf1000.orders LIMIT 0", tableName));
+            onTrino().executeQuery("SET SESSION scale_writers = true");
+            onTrino().executeQuery("SET SESSION writer_min_size = '4kB'");
+            onTrino().executeQuery("SET SESSION task_writer_count = 4");
+            onTrino().executeQuery("SET SESSION hive.target_max_file_size = '1MB'");
+
+            onTrino().executeQuery(format("INSERT INTO %s SELECT orderkey, orderstatus, totalprice, orderdate, clerk, shippriority, \"comment\", custkey, orderpriority FROM tpch.sf1000.orders LIMIT 100000", tableName));
+            assertThat(onTrino().executeQuery(format("SELECT count(*) FROM %s", tableName))).containsOnly(row(100000));
+            int numberOfCreatedFiles = onTrino().executeQuery(format("SELECT DISTINCT \"$path\" FROM %s", tableName)).getRowsCount();
+            assertEquals(numberOfCreatedFiles, 5, "There should be only 1 file created");
```

Comment:
wrong error message.

Btw: can we merge the tests and have the `boolean partitioned` argument?

Reply:
sure I can give this a try